diff --git a/dist/config/vite.main.config.js b/dist/config/vite.main.config.js
index 377604568e20c30e2bd6480551694d531b51396e..ce75aa0ba9a4dae7861ad019cf86a58486e18643 100644
--- a/dist/config/vite.main.config.js
+++ b/dist/config/vite.main.config.js
@@ -11,7 +11,7 @@ function getConfig(forgeEnv) {
             lib: {
                 entry: forgeConfigSelf.entry,
                 fileName: () => '[name].js',
-                formats: ['cjs'],
+                formats: ['es'],
             },
             rollupOptions: {
                 external: vite_base_config_1.external,
diff --git a/dist/config/vite.preload.config.js b/dist/config/vite.preload.config.js
index 20a3fe644eee3da63ca623efd00284da989258ff..64f7b0e4619a2f2306f5f0a66c8c2ca530091119 100644
--- a/dist/config/vite.preload.config.js
+++ b/dist/config/vite.preload.config.js
@@ -12,7 +12,7 @@ function getConfig(forgeEnv) {
                 // Preload scripts may contain Web assets, so use the `build.rollupOptions.input` instead `build.lib.entry`.
                 input: forgeConfigSelf.entry,
                 output: {
-                    format: 'cjs',
+                    format: 'es',
                     // It should not be split chunks.
                     inlineDynamicImports: true,
                     entryFileNames: '[name].js',
