diff --git a/dist/src/transform/index.js b/dist/src/transform/index.js
--- a/dist/src/transform/index.js
+++ b/dist/src/transform/index.js
@@ -615,6 +615,8 @@ function tspToProto(program, emitterOptions) {
         // Determine if the property type is an array
         if (isArray(property.type))
             field.repeated = true;
+        if (property.optional)
+            field.optional = true;
         return field;
     }
     /**
diff --git a/dist/src/write.js b/dist/src/write.js
--- a/dist/src/write.js
+++ b/dist/src/write.js
@@ -137,7 +137,7 @@ function writeVariant(decl, indentLevel) {
     return writeDocumentationCommentFlexible(decl, output, indentLevel);
 }
 function writeField(decl, indentLevel) {
-    const prefix = decl.repeated ? "repeated " : "";
+    const prefix = decl.repeated ? "repeated " : decl.optional ? "optional " : "";
     const output = prefix + `${writeType(decl.type)} ${decl.name} = ${decl.index};`;
     return writeDocumentationCommentFlexible(decl, output, indentLevel);
 }
